### DISCUSSION TITLE
No Sound out on  riverdi rvt70aqffwc00.

### DISCUSSION THREAD

**Discussion by MaWa53** [RudolphRiedel/FT800-FT813#60] [Q&A] [answered]
Hello FT/BT8x fans,
I run into a problem using sound output of my BT817 based riverdi rvt70aqffwc00. 
The display is driven from a RasPi4 via SPI and GPIO lines. A 8 Ohm loudspeaker is connect to the 2 pin output.
Volume is set to full 255.

I use this soundplayer class method:

> void SoundPlayer::play(effect_t effect, note_t note)  {
>       eveMemWrite8	(EVE_REG_VOL_SOUND, 255); 	
>       uint16_t val = (uint16_t) ((note << 8) | effect);
>       eveMemWrite16	(EVE_REG_SOUND, val);
>       eveMemWrite8	(EVE_REG_PLAY, 1); 	
> }
> 
> effect like CHACK	= 0x58
> note like NOTE_C4         = 0x3C

Does anybody have an idea what to check?

**Comment by RudolphRiedel** [comment]
I can't even test this as I do not have the speaker from Riverdi.

**Reply by MaWa53** [reply]
As far as I understand the schematic there is an 8 ohm loudspeaker amplifier behind the 2pin connector and the two soldering pads beside the connector. I bought a rvt50 as alternative but decided to go on with the 7 inch one. I will try to adapt my software to the 5 inch size and see if it's a problem with the 7 inch board or a software problem. But maybe someone else has the sound output running and can report his experience.

**Comment by RudolphRiedel** [comment, marked as answer]
I received the connectors and the contacts and built my own cable plus speaker.
And at first my RVT70HSBNWC00-B refused to make sounds.
What I initially missed though is that there is a schematic for the audio circuit in the datasheet and that GPIO2 from the BT817 enables the 
amplifier chip.

Now I added this at the end of TFT_init():
EVE_memWrite16(REG_GPIOX_DIR, 0x8004U); /* set GPIO2 to output to enable EVE4 audio output */
EVE_memWrite16(REG_GPIOX, 0x8004U); /* set GPIO2 to high to enable EVE4 audio output */
EVE_memWrite8(REG_VOL_SOUND, 0xff);

And to play this in  my touch function:
EVE_memWrite16(REG_SOUND, (85 << 8) | EVE_ALARM);
EVE_memWrite8(REG_PLAY, 1);

EVE_memWrite16(REG_SOUND, EVE_SILENCE);
EVE_memWrite8(REG_PLAY, 1);

**Reply by MaWa53** [reply]
Rudolf... you got it! Thanks a lot!  / Einen recht herzlichen Dank!

It's funny that sometimes you don't see a detail like the second connection from GPIO2 to AUDIO PD on the PAM8301. (:-(

Well Rudolf you found it and using your example code my display plays all sounds. 

I only wonder why no other complaints have come up because in **NONE** of the various documentations of Riverdi this important setting is mentioned.

There is still a very minor problem with the the amplifiers power supply. 
A clearly audible hum is emitted from the speaker when the amplifier is enabled which is **independent** from the volume.
In the later operating environment this hum will probably be drowned out by the noise of the fans and stepper motors but in my very quiet office writing my code the hum is a little bit distracting. I will try to add a large tantalum capacitor on _VDDP_ and _GND_ near the amplifier chip to see if that reduces the humming.

**BTW**: your example with EVE_ALARM and EVE_SILENCE brought me to the realization that there are two ways to play a sound depending on the feature that the sound is 'continuous' or not. 

- Continuous sounds need the SILENCE call to stop. 
- Noncontinuous sounds terminate by themselves.

Thanks again for your effort to find the error.

Regards
Manfred

**Reply by RudolphRiedel** [reply]
Well, I never used the audio out for anything else than test code and having an amplifier on the module is kind of new.

And now that you mention it, yes, I have a humming noise as well - only faint though since I am using a 50R speaker.
When I set GPIO2 to low the humming stops.
Switching the amplifier on or off results in a clicking noise that is even worse than the humming, even with silence for 2ms before switching off and 2ms delay after switching on before playing.

I guess I could get away with the humming but definately not with the clicking.

**Reply by MaWa53** [reply]
Today I checked the circuit with my oscilloscope. 
The **3.3V power** is very noisy with high frequency +/- 100mV spikes and a kind of random 'base' noise around +/- 20mV. 
Therefore the amplifier output produces the humming noise even if there is no sound signal on R6. There already is a larger smoothing capacitor on the 3.3V.

If you do not display anything the noise is slightly less. 

I therefore suppose the the 3.3V supply should be a bit 'stronger' with a smaller inner resistance and feed lines with a larger cross section.

I currently supply the display from the 5 and 3.3V regulators of by _Bigtreetech octopus controller board_ running a modified marlin firmware to control 6 stepper motors. The controller board is connected via serial I/O and the display via SPI & GPIO to a raspberry Pi 4 (which has an independent power supply) 
Everything for the display goes over the thin 0.5mm FFC band connector attached to a _RIBUS Click_ adapter board. 

**Not the optimal route for 96mA + 300mA @ 1W amplifier output!**

I intended to setup an individual 5V and 3.3V regulator for the display anyway, because I do not know how the power output of the controller board will look like when the steppers and fans are running. 

Maybe that will bring down the humming so the amp can remain powered on and no 'on' click will occur.
I will report my results.

**BTW**: Rudolf, where are you located? I live northwest of Hamburg.

Regards
Manfred

**Reply by MaWa53** [reply]
I forgot to mention - for those interested in details of the sound processing - a application note:

http://www.ftdichip.com/Support/Documents/AppNotes/AN_252%20FT800%20Audio%20Primer.pdf 

Quite interesting especially on page 8

**Reply by RudolphRiedel** [reply]
I took a quick peek at the datasheet and the board again and noticed that there
is a switching regulator near the audio amplifier.
I wonder why though as there is a second one on the other side which should be the one for the backlight.
Are they supplying the audio amplifier with a switching regulator from the backlight supply?

Regardless, try this:
EVE_memWrite16(REG_PWM_HZ, 9000U);

It does not stop the noise, but at least it seems to stop the modulation on the noise.

My controller board has two switching regulators and I am only using a 5cm FFC, power should not be an issue.

The application note should not be applicable since they went from simple PWM output in the FT8xx series to
a Sigma-delta converter in the BT81x series.


Und ohne Zusammenhang, ich komme aus dem Norden, Hamburg, Ratzeburg, Lübeck, und ich wohne und arbeite jetzt etwas nördlich von Braunschweig. :-)

**Reply by MaWa53** [reply]
A new problem with the display has become priority. I will issue it in an other thread.

Using your last suggestion the 'noise' has become bearable and I will live with it.

Especially since the speaker will be on the rear of a 45° slanted display stand. Main use of the sound is for an acoustic touch and user interface feedback and for alarm purpose. I will still change the 3.3V and 5V power supply. I will report if that brings any positive change.

Thank you again Rudolf.

**Reply by steve-4dproducts** [reply]
Thanks ever so much for the fantastic library. I was struggling today to get the sound working, but came across this post and it's working beautifully now on a Riverdi display. Thanks again.
